cmd/datactl/app/metering/export_pull_test.go

```diff
@@ -33,3 +40,94 @@ var _ = Describe("export_pull", func() {
 		Expect(sut).ToNot(BeNil())
 	})
 })
+
+var _ = Describe("export_pull_ilmt", func() {
+
+	Context("test whether ILMT API sending the expected response for date range difference 4 months", func() {
+		It("success", func() {
+
+			kubeConfigFlags := genericclioptions.NewConfigFlags(true).WithDeprecatedPasswordFlag()
+			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+			defer cancel()
+			o := exportPullOptions{
+				rhmConfigFlags: config.NewConfigFlags(kubeConfigFlags),
+				PrintFlags:     get.NewGetPrintFlags(),
+				sourceName:     "demo.ilmt.ibmcloudsecurity.com",
+				sourceType:     "ILMT",
+				startDate:      "2022-02-04",
+				endDate:        "2022-06-04",
+			}
+
+			o.rhmRawConfig, _ = o.rhmConfigFlags.RawPersistentConfigLoader().RawConfig()
+			o.Complete(nil, nil)
+			for name := range o.rhmRawConfig.Sources {
+				s := o.rhmRawConfig.Sources[name]
+				if s.Type.String() == o.sourceType {
+					count, response, err := o.IlmtPullBase(s, ctx, nil, nil)
+					Expect(err).To(Succeed())
+					Expect(997).To(Equal(count))
```

Comment:
I don't see any mocking done on the ILMT server here. So are you trying to call the actual server here?

Reply:
As discussed on Webex, when we are writing unit tests, all external dependencies should be mocked - That includes external services as well. So use mocking instead of calling a real server.